app/api/v2/managers/config_api_manager.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from app.utility.base_world import BaseWorld
+
+
+SENSITIVE_CONFIG_PROPS = frozenset([
+    'ability_refresh',
+    'api_key_blue',
+    'api_key_red',
+    'app.contact.gist',
+    'app.contact.tunnel.ssh.host_key_file',
+    'app.contact.tunnel.ssh.host_key_passphrase',
+    'app.contact.tunnel.ssh.user_name',
+    'app.contact.tunnel.ssh.user_password',
+    'auth.login.handler.module',
+    'crypt_salt',
+    'encryption_key',
+    'host',
+    'port',
+    'requirements',
+    'users'
+])
+
+
+def filter_keys(mapping, keys_to_remove):
+    filtered = {}
+
+    for key, val in mapping.items():
+        if key not in keys_to_remove:
+            filtered[key] = val
+
+    return filtered
+
+
+def filter_sensitive_props(config_map):
+    """Return a copy of `config_map` with top-level sensitive keys removed."""
+    return filter_keys(config_map, keys_to_remove=SENSITIVE_CONFIG_PROPS)
+
+
+def is_sensitive_prop(prop):
+    """Return True if the input prop is a sensitive configuration property."""
+    return prop in SENSITIVE_CONFIG_PROPS
+
+
+class ConfigUpdateNotAllowed(Exception):
+    def __init__(self, property, message=None):
+        super().__init__(message or f'Updating property is disallowed: {property}')
+        self.property = property
+
+
+class ConfigNotFound(Exception):
+    def __init__(self, config_name, message=None):
+        super().__init__(message or f'Config not found: {config_name}')
+        self.config_name = config_name
+
+
+class ConfigApiManager:
+    def __init__(self, data_svc, config_interface=None):
+        self._config_interface = config_interface or BaseWorld
+        self._data_svc = data_svc
+        self._log = logging.getLogger('config_api_manager')
+
+    def get_filtered_config(self, name):
+        """Return the configuration for the input `name` with sensitive fields removed."""
+        try:
+            config = self._config_interface.get_config(name=name)
+        except KeyError:
+            raise ConfigNotFound(name)
+        return filter_sensitive_props(config)
+
+    def update_main_config(self, prop, value):
+        if is_sensitive_prop(prop):
+            raise ConfigUpdateNotAllowed(prop)
+
+        if prop == 'plugin':
+            enabled_plugins = self._config_interface.get_config(
+                name='main',
+                prop='plugins'
+            )
+
+            if value not in enabled_plugins:
+                enabled_plugins.append(value)
+        else:
+            self._config_interface.set_config(
+                name='main',
+                prop=prop,
+                value=value
+            )
+
+    async def update_global_agent_config(self, sleep_min=None, sleep_max=None, watchdog=None, untrusted_timer=None,
+                                         implant_name=None, bootstrap_abilities=None, deadman_abilities=None):
+
+        set_config = self._config_interface.set_config
+
+        if sleep_min is not None:
```

Comment:
I don't like that we don't perform any validation on these values (aside from the datatype validation performed by marshmallow). I am gonna add some validation--just need to decide where the best place is do do that